src/util/json_expr.cpp

```diff
@@ -258,42 +274,41 @@ json_objectt json(
         type.id()==ID_c_bit_field?type.subtype():
         type;
 
-      bool is_signed=underlying_type.id()==ID_signedbv;
-
-      std::string sig=is_signed?"":"unsigned ";
+      languaget *lang;
+      if(mode==ID_unknown)
+        lang=get_default_language();
+      else
+      {
+        lang=get_language_from_mode(mode);
+        if(lang==NULL)
+          throw "unknown mode "+id2string(mode);
+      }
 
-      if(width==config.ansi_c.char_width)
-        result["c_type"]=json_stringt(sig+"char");
-      else if(width==config.ansi_c.int_width)
-        result["c_type"]=json_stringt(sig+"int");
-      else if(width==config.ansi_c.short_int_width)
-        result["c_type"]=json_stringt(sig+"short int");
-      else if(width==config.ansi_c.long_int_width)
-        result["c_type"]=json_stringt(sig+"long int");
-      else if(width==config.ansi_c.long_long_int_width)
-        result["c_type"]=json_stringt(sig+"long long int");
+      std::string type_string;
+      assert(!lang->from_type(underlying_type, type_string, ns));
+      result["type"]=json_stringt(type_string);
 
       mp_integer i;
-      if(!to_integer(expr, i))
-        result["data"]=json_stringt(integer2string(i));
+      assert(!to_integer(expr, i));
```

Comment:
I changed the commits: https://github.com/diffblue/cbmc/pull/889/commits/2e464eaff01831e4875280854384b9e95787a941 https://github.com/diffblue/cbmc/pull/889/commits/7f3adc16bbca2b8114412f9ceb558dff817f1a3a